sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -932,10 +998,14 @@ def get_buffer(buffer_id):
       return pcoll_buffers[buffer_id]
 
     def get_input_coder_impl(transform_id):
-      return context.coders[
-          safe_coders[beam_fn_api_pb2.RemoteGrpcPort.FromString(
-              process_bundle_descriptor.transforms[transform_id].spec.payload).
-                      coder_id]].get_impl()
+      coder_id = beam_fn_api_pb2.RemoteGrpcPort.FromString(
+          process_bundle_descriptor.transforms[transform_id].spec.payload).\
+        coder_id
+      assert coder_id is not None and coder_id != ''
```

Comment:
`coder_id` can be retrieved from transform payload most cases, except one following case. It returns `''` coder_id. 
This cause `test_checkpoint_sdf` test fails. How do we need to handle this case? Other SDF tests should be fixed now.

```
spec {
  urn: "beam:transform:sdf_process_sized_element_and_restrictions:v1"
  payload: "\n\215\014\n beam:dofn:pickled_python_info:v1\032\350\013eNp9ldt31FQUxnNmBihpKSCorRdE8TLVOlBaLkXFy0AFI0NNEaJSYiY50wQyyXzJCVKcEVAzxfv9vf0XXcu1fPDBfc5MVy9QXmbNuezz/fa39zm5Wyy7XhCGFVv+6rabcEdwWyy2uA5t41oYO15/hVklTdPkfxRyFE2UrCGaOBtlTZ7QAXNeA9tcp+W4Prfr3GlWROJEaSNOmmnFjROun4lnIh3bc+zoYKBs6RRt283Yy0Ju29hpDfQmnCCioW7toGEriV2ephh8GHEji1wRxHTmUNlnKrwae/ySRNyVY9jE7rJRNDRjm1EyJqszC1qn2C4sayvaddYp3Rqn/4V2aYUlp9rF5WKbtUuNwgpj2n1tubSiMa1d8tgJrVPCnitau3iZZjxtTsOemsG62Fu29pGgmyUJj4Sd8FQkgcLBY9Y2WkmFkwjss3ZK15JF2w2doIn91nYaO60Wjzw8rhaDiPJp0iF4QuDJsrI55WEDI8oCHvYWR5XeOh2b/HVv8ARPqX0itnkzEHjaKva48IzAs0ou9RqV1iIO+IMXNFYd1kolprMhNswG2FABz1m712RsN84iQYcezPH8mMALJg65tl3PglDIuuirBdL9wUcW2zbXSGXhZ53Eaep4ceweXurg5bK1f1M2VOqbgUfKr/ha2S9Ze9c31+zqYvmRous0VyN0jOV4tYPXyv5OX1ceBlEgAifcULNxn3qI2qVA7VKgdpmrDgltmXmszVYKVPTXa4bWReUB6n4NUhy2BmnpYqORcmE60QLHkS4m/BF/tIuj/gF/3Pi7WtAYJsfI1SkTx7ZwdbxW6+B4TeCEiZPWsCxl/372qz29gbRIpLokJU5GnKdqOd7wD1sjazBrtJf6R7zZxVv+iAJe78HpLt4m0mnjH0X6jiR918R7W5BOS9IqkZ4xcbbX8q2QGnCmDyhvXtFgBLi9XViQcO8T3Lkx6Yl/egnnSWvG+Je0CvhAahkmPtxCa0ZqXSCtmomL1p5NRUiD2xyzG2QLJDvQu+wk/BEJm72LJbfO5bi0SvExUcwa/6mML0uKKyasLShmJcUnRPGpic/UpbNtL6bNuFqzdqlhGsYicpo8pcn5eVzLBGwTn1PXO+qBaslrYAce6tYhGq5vWLlSfqDjx+Bm9Rzewx5Al9MAvJ84o8QZJV6tDgitrY3KxBuU+II1KtFagXsj5J696aKn8HMEyhvJjes5bqx/NMJedBb14jeHo3nhGlPmRdK82ERrC/Oa0jyQeYmJ1DpIh04frfOTE8ecqfqkNzVVn6jzqSMnjx6ZPO5OT7j82AREjszEzRxfmLiVY7GD2xT/pYm2f7Xmk70d1cPqA2X7QSRSfPUwn2RinshaIX3G7qiQ8xflV+KcCrmrZoKolQl1Uop76ssWZ2Jt6mvlSZwEC0GEb5bwrcpgw1tE+xRDxeP0GDkiJm/zJXRNLNVq87i/hO9MfE8N8YOJH6khfurg5wf4f/HV5l+X8JuJ39P6PP7o4M8apf1X5X8g4Xjr0\001:\037ref_Coder_FastPrimitivesCoder_4"
}
inputs {
  key: "0"
  value: "ref_PCollection_PCollection_3_split"
}
outputs {
  key: "None"
  value: "ref_PCollection_PCollection_4"
}
unique_name: "SDF/Process"
environment_id: "ref_Environment_default_environment_1"
```

Reply:
OK, in the SDF case we can get the coder of the preceding transform (the one that produces its input `ref_PCollection_PCollection_3_split`) which should always be a RemoteGrpcPort.